lib/openhab/core/items/semantics.rb

```diff
@@ -390,6 +397,67 @@ def points(*point_or_property_types)
           result.delete(self)
           result
         end
+
+        # @deprecated OH3.4 - this check is only needed for OH3.4
+        if org.openhab.core.semantics.SemanticTags.respond_to?(:add)
+
+          #
+          # Adds custom semantic tags.
+          #
+          # @return [Semantics::Tag] The added semantic tag class
+          #
+          # @overload self.add(**tags)
+          #   Quickly add one or more semantic tags using the default label, empty synonyms and descriptions.
+          #
+          #   @param [kwargs] **tags One or more `tag` => `parent` pairs
+          #   @return [Boolean] true if all tags were added successfully
+          #
+          #   @example Add one semantic tag `Balcony` whose parent is `Semantics::Outdoor` (Location)
+          #     Semantics.add(Balcony: Semantics::Outdoor)
+          #
+          #   @example Add multiple semantic tags
+          #     Semantics.add(Balcony: Semantics::Outdoor,
+          #                   SecretRoom: Semantics::Room,
+          #                   Motion: Semantics::Property)
+          #
+          # @overload self.add(label: nil, synonyms: "", description: "", **tags)
+          #   Add a custom semantic tag with extra details.
+          #
+          #   @example
+          #     Semantics.add(SecretRoom: Semantics::Room, label: "My Secret Room",
+          #       synonyms: "HidingPlace", description: "A room that requires a special trick to enter")
+          #
+          #   @param [String,nil] label Optional label. When nil, infer the label from the tag name,
+          #     converting `CamelCase` to `Camel Case`
+          #   @param [String] synonyms A comma separated list of synonyms for this tag.
+          #   @param [String] description A longer description of the tag.
+          #   @param [kwargs] **tags Exactly one pair of `tag` => `parent`
+          #   @return [Boolean] true if the tag was added successfully
```

Comment:
It was meant to return true/false because it can create multiple tags. Overall I'm not sure... I guess it can return an array of created tags, or an empty array if none were created, that might be more useful than just a boolean, wdyt?

Reply:
I've changed it to return an array of the created tags. So when it's `#empty?` it means none were created.